pkgs/by-name/cu/cups-idprt/package.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenvNoCC
+, lib
+, fetchurl
+, autoPatchelfHook
+, cups
+, unzip
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "cups-idprt";
+  version = "1.4.2";
+
+  src = fetchurl {
+    name = "idprt_tspl_printer_linux_driver_v${version}.zip";
+    url = "https://www.idprt.com/prt_v2/files/down_file/id/131/fid/431.html"; # NOTE: This is NOT an HTML page, but a ZIP file
+    hash = "sha256-xY5E1RUNQwK0BO+0JMT7oHCAvSe79t7UU65IOzRQs/Q=";
+  };
+
+  buildInputs = [
+    cups
+  ];
+  nativeBuildInputs = [
+    autoPatchelfHook
+    unzip
+  ];
+
+  installPhase = let
+    arch = builtins.getAttr stdenvNoCC.hostPlatform.system {
+      x86_64-linux = "x64";
+      x86-linux = "x86";
+    };
+  in ''
+    runHook preInstall
+    mkdir -p $out/share/cups/model $out/lib/cups/filter
+    cp -r filter/${arch}/. $out/lib/cups/filter
+    cp -r ppd/. $out/share/cups/model
```

Comment:
Does the extra `/.` in `cp` do anything?

Reply:
Yes.

```
cp -r ppd/ $out/share/cups/model
```

copies the `ppd` folder to `$out/share/cups/model`, instead of its contents.